File: DSA/DataStructures/Array/6. ValidAnagram/Solution2.java
// https://leetcode.com/problems/valid-anagram/description/

class Solution {
    public boolean isAnagram(String s, String t) {

        if(s.length()!=t.length()) return false;
       Map<Character, Integer> freqMap = new HashMap<>();

       for(char c : s.toCharArray()) {
            freqMap.put(c,(freqMap.getOrDefault(c,0)+1));
       }

       for(char c : t.toCharArray()) {
            int count = freqMap.getOrDefault(c,0);

            if(count==0) {
                return false;
            }
            freqMap.put(c,count-1);
       }
        return true;
        
    }
}

File: DSA/DataStructures/Array/8. MergeTwoArray/Solution2.js
// https://leetcode.com/problems/merge-sorted-array/description/

var merge = function(nums1, m, nums2, n) {
    
    let l = m+n-1
    m--
    n--

    while(n>=0) {

        if(m>=0 && nums1[m]>nums2[n]) {
            nums1[l] = nums1[m]
            m--
        } else {
            nums1[l] = nums2[n]
            n--
        }
        l--
    }
};